### DISCUSSION TITLE
Explore adding flow-storm debugger support to Calva

### DISCUSSION THREAD

**Discussion by bpringe** [BetterThanTomorrow/calva#2453] [Ideas]
I'm not sure if this is something that should be integrated into Calva, or left to devs to know about and add to their own REPLs, but it's really interesting. It seems in some ways it goes further than Cider's debugger (and works a bit differently - it's a trace debugger) - supports Clojure _and_ ClojureScript, stepping forward _and backward_ in time, inspecting large values, and more). See the github and feature list here:

https://github.com/jpmonettas/flow-storm-debugger#features

A defn podcast episode with an interview with the creator: https://soundcloud.com/defn-771544745/67-juan-monetta

**Comment by riotrah** [comment]
whoa

**Comment by bpringe** [comment]
A 2.0 version of flow-storm is now out but it seems only Clojure is supported so far. I believe ClojureScript _was_ supported before, but that support is planned for the new version as well.

**Comment by bpringe** [comment]
There's now a Cider integration for FlowStorm, called [CiderStorm](https://github.com/jpmonettas/cider-storm). 

Slack message: https://clojurians.slack.com/archives/C06MAR553/p1686673281768159

From the Slack message:
> Editors can use this nrepl middleware https://github.com/jpmonettas/flow-storm-debugger/blob/master/src-inst/flow_storm/nrepl/middleware.clj#L199-L275 to communicate with FlowStorm as CiderStorm is doing.
